factorial/factorial.c

```diff
@@ -0,0 +1,45 @@
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+// Signatures of functions defined later
+int validateInput(int, char **);
+uintmax_t factorial(uintmax_t);
+
+// Entrypoint
+int main( int argc, char *argv[] )  {
+    uintmax_t value = validateInput(argc, argv);
+    printf("Input value: %lu\n", value);
+    uintmax_t result = factorial(value);
+    printf("     Result: %lu\n", result);
+    return 0;
+}
+
+// Calculate factorial
+uintmax_t factorial(uintmax_t value) {
+    if (value < 2) {
+        return 1;
+    }
+    uintmax_t counter = value;
+    uintmax_t result = value;
+    while (counter > 1) {
+        result = result * (counter - 1);
+        counter--;
+    }
+    return result;
+}
+
+// Validate input parameters
+int validateInput(int argc, char *argv[]) {
```

Comment:
your method signature here has a (potentially) lossy implicit type conversion.  The function signature specifies that it returns `int` but the return statement returns a value of type `uintmax_t`.  I think it would be best if they were both the same type just in case `int` has less precision on the target platform than uintmax_t would have.

Reply:
Good catch.  I intended it to be int. It didn't make sense to use unitmax_t because our input values can never get that high.  Will fix.

Reply:
fixed